### DISCUSSION TITLE
Can I add ENV variables in the vercel project that doesn't belongs to my vercel account?

### DISCUSSION THREAD

**Discussion by adityamittal3107** [vercel/community#4920] [Help]
### Summary

Hi team,
I was trying to create a vercel integration in which I was using vercel deploy button to deploy a project in the user's vercel account 
After deploying the project I have got the project info in the redirect url. But now if I am trying to add env variables in that project it is responding with error : {
  "code": "not_found",
  "message": "Project not found."
}
Is there any way in which I can add env variables on a vercel project that is on other user's account?

### Steps to Reproduce

1. Use Vercel deploy button to deploy a project in customer's application.
2. Use Vercel API - `https://api.vercel.com/v9/projects/${projectIds}/env?upsert=true&teamId=${teamId}` to add env variables on the project deployed above.

**Comment by amyegan** [comment]
Hi Aditya! You can only modify environment variables for the projects on your own account or team.

**Reply by adityamittal3107** [reply]
Hi @amyegan but we can add Env variables to the project that customer select while starting integration so is there any way to do this with newly added project in the middle of the integration?

**Reply by amyegan** [reply]
If you're using a [deploy button](https://vercel.com/docs/deployments/deploy-button), the environment variables are added when the project is created and can only be modified by the owner during and after project creation. 

If you created an [integration](https://vercel.com/docs/integrations/create-integration), it could update environment variables if the integration has "Project Environment Variables" read/write access enabled. 

The project owner always has the option to modify environment variables themselves as well.

**Reply by adityamittal3107** [reply]
How can I add the environment variables when the project is created by vercel deploy button? Here it says the user has to pass the value we can only mention the keys of environment variables. Is there any way in which we can prefill the values of environment variables?
https://vercel.com/docs/deployments/deploy-button/environment-variables

**Reply by adityamittal3107** [reply]
And in the integration I have  "Project Environment Variables" read/write access enabled but it allows me to add env variables on only those projects which user has selected before starting integration.
<img width="486" alt="image" src="https://github.com/vercel/community/assets/114516106/74fb7f16-7731-4b1c-a2bb-c3e8f2652103">
